pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -693,6 +693,11 @@ func buildSidecarOutboundTCPListenerOptsForPortOrUDS(listenerMapKey *string,
 	var destinationCIDR string
 	if len(listenerOpts.bind) == 0 {
 		svcListenAddress := listenerOpts.service.GetAddressForProxy(listenerOpts.proxy)
+		// if listenerOpts.service is not the common k8s svc and headless svc, it may be a svc converted from service entry, etc.
+		// It is hard to detect IP family for service entries, so use the current proxy's IP Mode.
+		if listenerOpts.service.Attributes.ServiceRegistry == provider.External && listenerOpts.proxy.IsIPv6() {
```

Comment:
shouldn't this only be when it is originally `constants.Unspecified` we switch it to `constants.UnspecifiedIPv6`? Seems like this would wipe out any VIPs

Reply:
Hi @howardjohn, `listenerOpts.service.Attributes.ServiceRegistry == provider.External` means that it's not the normal Istio service in service mesh, it should be service entry. So there is no any ClusterVIP(I understand you mentioned `VIP` means `ClusterVIPs`) would be impacted. And we should switch `constants.Unspecified` into `constants.UnspecifiedIPv6` if the listenersOpts's proxy only support IPv6 and there is no any other address bind.  Hope I explains it clearly.

Reply:
ServiceEntry can have a VIP from the `addresses` field

Reply:
Yeah, so you worry on how about if there are IPv4 and IPv6 in service entry's addresses? 
Based on current implementation, `constants.Unspecified` is the only choice, no matter it is correct or not. However, it's okay if we follow the `listenerOpts.proxy` of the service, the available IP family may be elected at least.

Reply:
Yeah @howardjohn, so you worry on how about if there are IPv4 and IPv6 in service entry's addresses? 
Based on current implementation, `constants.Unspecified` is the only choice, no matter it is correct or not. However, it's okay if we follow the `listenerOpts.proxy` of the service, the available IP family may be elected at least.
Moreover, `constants.UnspecifiedIPv6` can be available for both IPv4 and IPv6 by default.

Reply:
I don't understand how `constants.Unspecified` can be the only choice..? I don't think that is accurate. A service entry is a superset of the `Service` API (more or less), it can have VIPs